esp-wifi/src/lib.rs

```diff
@@ -94,6 +94,16 @@
 // toolchain doesn't know about that lint, yet)
 #![allow(unknown_lints)]
 #![allow(non_local_definitions)]
+#![cfg_attr(
+    not(any(feature = "wifi", feature = "ble")),
```

Comment:
esp-wifi isn't our highest quality code yet, and trying to figure out what exactly is unused, and adding `cfg`s on them felt like excessive nonsense to do.